app/models/user.rb

```diff
@@ -53,6 +53,9 @@ class User < ApplicationRecord
   serialize     :settings, Hash   # Implement settings column as a hash
   default_value_for(:settings) { Hash.new }
 
+  default_value_for :failed_login_attempts, 0
+  virtual_column :locked?, :type => :boolean
```

Comment:
Does this need to be a virtual column?  I don't think we need to report on it.

Reply:
Or do you need this for API?

Reply:
Well, I didn't start working on the API yet, but @gtanzillo wants some kind of UI way to allow the admin to unlock a user's account. As we're moving to an API-driven UI, I thought eventually we will need it there for sure.

I didn't start working on the UI part yet, so I don't know if the unlock there will be API-driven or not, but I guess having this won't hurt. But ofc I can remove it if you don't like it, just let me know.

Reply:
I don't see why we'd need account unlocking if it happens automatically after N minutes. 

However, I could see it being necessary in an API driven UI to be able to see the currently locked accounts. However, even then, it might be better as a real column, because you'll probably want to query for that, and being a virtual column will make querying difficult.

I'm leaning towards removing this at the moment until we need it, and leaning towards a dedicated column when we actually need it.

Reply:
@jvlcek Are you expecting an admin would curl an API or something?  I guess I don't see how that's much different than ssh'ing and doing a console based change.  Also if the admin account *itself* is locked out, they wouldn't be able to use the API anyway and we would need a console based approach regardless.

Reply:
I'm going with the console-only unlock approach.

Reply:
@skateman as discussed I'm good with your approach. 👍